pkg/service/repair/controller.go

```diff
@@ -177,57 +173,51 @@ func (c *rowLevelRepairController) TryBlock(hosts []string) (bool, allowance) {
 	return true, a
 }
 
+func (c *rowLevelRepairController) parallelLimit() int {
+	p := c.intensity.Parallel()
+
+	if p <= defaultParallel {
+		return c.intensity.MaxParallel()
+	}
+
+	return p
+}
+
 func (c *rowLevelRepairController) shouldBlock(hosts []string, intensity float64) bool {
 	// ALLOW if nothing is running
 	if !c.Busy() {
 		return true
 	}
 
-	// DENY if any host has max nr of jobs already running
+	// DENY if any host has a job running already. We don't allow more than a single repair job per host.
 	for _, h := range hosts {
-		if c.jobs[h] >= c.limits[h].Default {
+		if c.jobs[h] >= 1 {
+			//fmt.Printf("all jobs: %d, Too many jobs for %s (%d). Blocking.\n", c.allJobs, h, c.jobs[h])
 			return false
 		}
 	}
 
-	// DENY if any host already runs at the full intensity.
-	// This may happen if we decrease intensity, for a while we need to reduce
-	// number of jobs to reduce number of busy ranges.
-	ranges := c.rangesForIntensity(hosts, intensity)
-	for _, h := range hosts {
-		var (
-			r  = c.busyRanges[h] + ranges
-			ok bool
-		)
-		switch {
-		case intensity == maxIntensity:
-			ok = r <= c.limits[h].Max
-		case intensity <= 1:
-			ok = r <= c.limits[h].Default
-		default:
-			ok = r <= int(intensity)*c.limits[h].Default && r <= c.limits[h].Max
-		}
-		if !ok {
-			return false
-		}
+	// DENY (Sanity) if allJobs + 1 > MaxParallel.
+	// However, this should never happen due to "up to 1 repair task per replica" and the fact that MaxParallel is
+	// calculated the way that allows up to 1 repair task per host.
+	if c.allJobs+1 > c.intensity.MaxParallel() {
+		//fmt.Println("Too high parallelism (vs MaxParallel). Blocking.")
+		return false
 	}
 
-	// DENY if there is too much parallel replicas being repaired
-	if parallel := c.intensity.Parallel(); parallel != defaultParallel {
-		hash := replicaHash(hosts)
-		if _, ok := c.busyReplicas[hash]; !ok {
-			l := len(c.busyReplicas) + 1
-			if l > parallel || l > c.intensity.MaxParallel() {
-				return false
-			}
-		}
+	// DENY if allJobs is greater than the configured parallelism
+	if c.allJobs+1 > c.parallelLimit() {
+		//fmt.Println("Too high parallelism (vs Parallel). Blocking.")
+		return false
 	}
 
 	// DENY if all workers are busy
 	if c.allJobs == c.maxWorkerCount {
+		//fmt.Println("All workers are busy. Blocking.")
```

Comment:
instead of comments isn't there some debug flag or logger (debug) level where you can put this?

Reply:
I didn't want to create too much "noise" in this PR.
I was planning to do this in the followup patches.
If it bothers you I can drop all this patch.